cmd/upgrade/accountroles/cmd.go

```diff
@@ -95,15 +97,14 @@ func run(cmd *cobra.Command, argv []string) error {
 
 	isInvokedFromClusterUpgrade := false
 	skipInteractive := false
-	if len(argv) >= 2 && !cmd.Flag("prefix").Changed {
-		args.prefix = argv[0]
-		aws.SetModeKey(argv[1])
+	var cluster *v1.Cluster
+	if len(argv) == 2 && !cmd.Flag("prefix").Changed {
```

Comment:
This will not work if the user does 

rosa upgrade accoount-roles --cluster-id

Reply:
This is only when calling from within `upgrade cluster`